internal/databases/mysql/binlog_push_handler.go

```diff
@@ -37,21 +36,28 @@ func HandleBinlogPush(uploader internal.UploaderProvider, untilBinlog string, ch
 	binlogsFolder, err := getMySQLBinlogsFolder(db)
 	tracelog.ErrorLogger.FatalOnError(err)
 
-	binlogs, err := getMySQLSortedBinlogs(db)
+	binlogs, err := getMySQLBinlogs(db)
 	tracelog.ErrorLogger.FatalOnError(err)
 
 	lastBinlog := lastOrDefault(binlogs, "")
-	if untilBinlog == "" || untilBinlog > lastBinlog {
+	if untilBinlog == "" || BinlogNum(untilBinlog) > BinlogNum(lastBinlog) {
 		untilBinlog = lastBinlog
 	}
 
 	var binlogSentinelDto BinlogSentinelDto
 	err = FetchBinlogSentinel(rootFolder, &binlogSentinelDto)
-	// copy MySQLBinlogSentinel to cache:
-	cache := getCache()
 	if err == nil && binlogSentinelDto.GTIDArchived != "" {
 		tracelog.InfoLogger.Printf("fetched binlog archived GTID SET: %s\n", binlogSentinelDto.GTIDArchived)
 	}
+	cache := getCache()
+	if len(binlogs) > 0 && cache.LastArchivedBinlog != "" {
```

Comment:
`len(binlogs) > 0` is always `true` here - when `binlogs` slice is empty - it will crash in line 43

Reply:
`len(binlogs) > 0` is always `true` here - when `binlogs` slice is empty - it will crash in line 43